frontend/web/src/app/components/digital-print/digital-print-list/digital-print-list.component.ts

```diff
@@ -0,0 +1,162 @@
+import {AfterViewInit, Component, EventEmitter, Input, OnInit, Output, ViewChild} from '@angular/core';
+import {Router} from '@angular/router';
+import {BsModalRef, BsModalService} from 'ngx-bootstrap';
+import {DigitalPrint} from '../../../model/digital-print';
+import {DigitalPrintService} from '../../../services/digital-print/digital-print.service';
+import {ErrorService} from '../../../services/error.service';
+declare var $: any;
+
+@Component({
+  selector: 'app-digital-print-list',
+  templateUrl: './digital-print-list.component.html',
+  styleUrls: ['./digital-print-list.component.scss'],
+  providers: [DigitalPrintService, ErrorService]
+})
+export class DigitalPrintListComponent implements OnInit, AfterViewInit {
+  private modalRef: BsModalRef;
+  @ViewChild('modalDelete') modalDelete;
+  private toBeDeleted: number = null;
+
+  private table;
+  @Input('digitalPrint') digitalPrints: DigitalPrint [];
+  @Output() selected: EventEmitter<DigitalPrint> = new EventEmitter<DigitalPrint>();
+
+
+  constructor(private digitalPrintService: DigitalPrintService,
+              private errorService: ErrorService,
+              private router: Router,
+              private modalService: BsModalService) {
+  }
+
+  ngOnInit() {
+    this.table = $('#table');
+  }
+
+  ngAfterViewInit(): void {
+    this.loadAndPopulate();
+  }
+
+  loadAndPopulate() {
+    /* get users */
+    this.digitalPrintService.getAllDigitalPrint().subscribe(
+      (digitalPrints) => {
+        this.digitalPrints = digitalPrints;
+
+        this.populate();
+        this.prepareTriggers();
+      }, (error) => {
+        alert(error.verbose_message);
+      },
+    );
+  }
+
+  private populate() {
+    const _self = this;
+    /*manually generate columns*/
+    const columns = [ {
+      title: 'ID',
+      field: 'id',
+      sortable: true,
+    }, {
+      title: 'Magics File',
+      field: 'magicsPath',
+      sortable: true,
+    }, {
+      field: 'operate',
+      title: '<span class="glyphicon glyphicon-cog">',
+      align: 'center',
+      events: {
+        'click .edit': function(e, value, row, index) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Expected method shorthand in object literal ('{'click .edit'() {...}}'). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS3498)